lib/src/socket.dart

```diff
@@ -508,10 +516,6 @@ class PhoenixSocket {
   void _onSocketData(message) => onSocketDataCallback(message);
 
   void _onSocketError(dynamic error, dynamic stacktrace) {
-    if (_socketState == SocketState.closing ||
```

Comment:
@matehat 
While testing the original change, I noticed that this check was preventing the socket from being reconnected, since `_onSocketClosed()` is called after it. The scheduling of the socket reconnection happens at that point.

Does it make sense?